In [8]:
import math

import math
import matplotlib.pyplot as pl

a,b,fo,f1,xo=[],[],[],[],[]
a=[[-1.1,10],[-2,17.4]]
b=[120,174]
x1,x2=0,0

def inicializar_Matriz(fil,col):
    
    mat= [None] * fil
    
    if(fil>0):
        for i in range(fil):
            mat[i] = [None] * col
            
    return mat

def func1_X1(x): 
    y=(10*x -120)/1.1
    return y

def func1_X2(x): 
    y=(1.1*x +120)/10
    return y

def func2_X1(x): 
    y=(17.4*x -174)/2
    return y

def func2_X2(x): 
    y=(174+2*x)/17.4
    return y

def ecua1(x1,x2):
    
    y=(-1.1*x1)+(10*x2)
    return y

def ecua2(x1,x2):
    
    y=-2*x1 +17.4*x2
    return y


def metodo_grafico():
    
    global x1,x2
    n,i=0,0
    
    while(i<5000):
        
        if(func1_X1(n)-func2_X1(n)<=0.00001):

            x1=func1_X1(n)
            x2=func2_X2(x1)
            
        fo.append(func1_X1(n))
        f1.append(func2_X1(n))
        xo.append(n)
        n+=0.1
        i+=1
    
    if(x1!=0 and x2!=0):
        print("x1=",x1)
        print("x2=",x2)
        print("(-1.1*x1)+(10*x2) =",ecua1(x1,x2))
        print("(-2*x1)+(17.4*x2) =",ecua2(x1,x2))
        
def graf():
    
    pl.ylabel('X1')
    pl.xlabel('X2')
    
    pl.plot(x2,x1,"ro",color="black",linewidth=20)
    pl.plot(xo,fo,color="green",label="Primera Ecuacion",linewidth=2.5, linestyle="-")
    pl.plot(xo,f1,color="pink",label="Segunda Ecuacion",linewidth=2.5, linestyle="-")
   
    pl.axis([40,60,200,410])
    pl.grid(True)
    pl.show()

def cal_determinante_dos_por_dos(a):
    return (a[0][0]*a[1][1])-(a[0][1]*a[1][0])

def eliminacion_incg(a,b):
    
    global x1,x2
    mat=inicializar_Matriz(len(a),len(a))
    sum=inicializar_Matriz(len(a)+1,0)
    
    cons,aux,auxx,i,j,x,z,cf=0,0,0,0,0,0,0,(len(a))

    while i<len(mat):
        x-=1
        cf=a[x][0]
        while j<len(mat[0]):
            mat[i][j]=a[i][j]*cf
            j+=1
        
        j=0
        
        cons=b[i]
        
        if(i>0):
            auxx=auxx-(cf*cons)
        else:
            auxx=auxx+(cf*cons)

        j=0
        i+=1
    
    sum[0]=mat[0][0]-mat[1][0]
    sum[1]=mat[0][1]-mat[1][1]
    sum[i]=(auxx)
    
    x2=(sum[2]/sum[1])
    x1=func2_X1(x2)
    print(mat,sum)
    print("x1=",x1)
    print("x2=",x2)
    print("(-1.1*x1)+(10*x2) =",ecua1(x1,x2))
    print("(-2*x1)+(17.4*x2) =",ecua2(x1,x2))
        
metodo_grafico()
print("Determinanre_A:",cal_determinante_dos_por_dos(a))
graf()
eliminacion_incg(a,b)

x1= 404.5454545454594
x2= 56.49947753396086
(-1.1*x1)+(10*x2) = 119.99477533960328
(-2*x1)+(17.4*x2) = 174.0
Determinanre_A: 0.8599999999999994
[[2.2, -20], [2.2, -19.14]] [0.0, -0.8599999999999994, -48.599999999999994]
x1= 404.6511627906979
x2= 56.51162790697678
(-1.1*x1)+(10*x2) = 120.0
(-2*x1)+(17.4*x2) = 174.0


Dado el sistema de ecuaciones siguiente
–1.1x1 + 10x2 = 120
–2x1 + 17.4x2 = 174
a) Resuélvalo gráficamente y compruebe el resultado con la
sustitución en las ecuaciones.
b) Sobre la base de la solución gráfica, ¿qué se espera con
respecto de la condición del sistema?
c) Calcule el determinante.
d) Resuelva por medio de la eliminación de incógnitas.

Solucion

a)
x1= 404.5454545454594
x2= 56.49947753396086
(-1.1*x1)+(10*x2) = 119.99477533960328
(-2*x1)+(17.4*x2) = 174.0

b)El sistema esta mal condicionado, al observar la grafica es dificil encontrar el punto de interseccion de las pendientes visualmente

c)Determinanre_A: 0.8599999999999994

d)
x1= 404.6511627906979
x2= 56.51162790697678
(-1.1*x1)+(10*x2) = 120.0
(-2*x1)+(17.4*x2) = 174.0